config/ola.m4

```diff
@@ -110,19 +131,36 @@ AC_DEFUN([PLUGIN_SUPPORT],
 
   AC_ARG_ENABLE(
     [$1],
-    AS_HELP_STRING([--disable-$1], [Disable the $1 plugin]))
+    AS_HELP_STRING([--disable-$1], [Disable the $1 plugin]),
+    [],
+    [eval $enable_arg=auto])
 
   eval enable_plugin=\$$enable_arg;
+
+  # Input validation.
+  if test "${enable_plugin}" != "yes" -a \
+          "${enable_plugin}" != "no" -a \
+          "${enable_plugin}" != "auto"; then
+    AC_MSG_ERROR([Invalid value for --enable-${plugin_key}/\
+--disable-${plugin_key}. Valid values are yes, no and auto.])
+  fi
+
+  # If dependencies are not met...
   if test "$3" == "no"; then
+    # ...and the user has explicitely requested this plugin to be enabled,
+    # error out.
+    if test "${enable_plugin}" = "yes"; then
+      AC_MSG_ERROR([Dependencies for plugin ${plugin_key} are not met.])
+    fi
+
+    # Otherwise, force the plugin disabled silently.
     enable_plugin="no";
   fi
 
   # If we've disabled all plugins, disable this one unless it has been
-  # explicitly enabled
-  if test "${enable_all_plugins}" == "no"; then
-    if test "${enable_plugin}" != "yes"; then
-      enable_plugin="no";
-    fi
+  # explicitly enabled.
+  if test "${enable_all_plugins}" = "no" -a "${enable_plugin}" != "yes"; then
```

Comment:
> If we've got the third state of auto, we either need an output plugin to check, or need to set auto's to no if disabling all plugins.

I am not sure I understand. That's exactly what this line is doing. If the plugin's state is auto and we want to disable all plugins, we override the state to "no" so the plugin doesn't get built.